Implemented abstract interactable behaviour

mprysia opened the pull request:
Created an abstract class to contain all types of handlers for
interactable objects. For now interactables only need to be able to
respond to selection. As a simple example I added an initial
implementation of pushable. This will prompt the PlayerManager to switch
to a PushController on the next frame.

mprysia committed:
Implement interactable object behaviour

Created an abstract class to contain all types of handlers for
interactable objects. For now interactables only need to be able to
respond to selection. As a simple example I added an initial
implementation of pushable. This will prompt the PlayerManager to switch
to a PushController on the next frame.

Maroski commented on Assets/Scripts/PlayerManager.cs:
```diff
@@ -35,6 +36,12 @@ private void Start()
 
     private void Update()
     {
+        // Context switch was requested
```
Add a comment to ensure this code is always before any interactions with a controller (to prevent issues with using a different controller midway through this method)

Maroski commented:
Add that comment as above, and add the changes you made to allow returning to the PlayerController. We've reviewed that code together on your laptop, so just push that, and merge this pull request so this doesn't block you from #6 and #4

mprysia committed:
Fix logic in player controller

The handlers that were being called in player manager (the ones that the
player controllers implement) were not exactly what we wanted. It would
make most sense to have a handler called every time the object the player
looks at changes (OnTargetChange) and one for everytime the player stares
at the same object for an extended period of time (OnHover). This solves
issue #9.

This also addresses the remarks of PR #3.

mprysia commented:
@Maroski feel free to delete the branch once you finish looking over the latest commit.
